### ISSUE TITLE
Question about the losses in PyTorch

### ISSUE THREAD

**Issue by jinyu121** [shepnerd/inpainting_gmcnn#52]
I noticed that the PyTorch version is not just the same as the paper, and have some strange points.
For example:

1. `self.wganloss = WGANLoss()` is never used, which is used in the paper
2. The `StyleLoss` class is never used
3. The `ContentLoss` class is never used (Seems that these two losses are merged into `IDMRFLoss`?)
4. The `TVLoss` class is never used
5. The re-construct loss and auto-encoder loss are both `L1Loss`, seems that there are no these two losses in the paper

The `VGG19` class is also never used, replaced by vgg19 in torchvision. (Is there any difference between the two VGG?)

And, if in stroke mode, `self.gt_local = self.gt`, `self.completed_local = self.completed`. So the global and local net in `self.netD` are just receiving the same input, do the same thing, and produce the same output. So... can I just use only one net in netD?

**Comment by shepnerd** [comment]
In the PyTorch implementation, as claimed in the disclaimer (README.md), we use a different GAN loss (wgan hinge loss with spectral normalization) instead of wgan-gp. 
For your question 2-4, those are redundant codes for this task, which are to be cleaned. Thanks for pointing out. 
For your question 5, the used reconstruction loss is intrinsically l1 loss. The difference between our used reconstruction loss and the common one is ours are incorporated spatial variance.
The code about the VGG19 class is also unnecessary. It will be removed. Thanks for pointing out.
For your last question, yes, using one discriminator is fine.
